### PR TITLE
chore: drop support for older Node.js versions

### LINKED ISSUE #1014: Drop support for Node.js 6 + 11
Checklist

- [x] Remove unsupported versions from package.json `engines` field (should be: `^8.6.0 || 10 || 12`)
- [ ] Mention minimum required version in docs to be v8.6.0
- [x] Remove forced upgrade of npm from Travis and Jenkins
- [x] Remove Node.js 6 and 11 tests from Travis and Jenkins
- [x] Bump the AppVeyor version of Node.js to be v10

### PR DESCRIPTION
**Pull request by watson**
Node.js versions older than 8.6.0 + the 11.x branch are no longer supported.

~~Depends on #1387~~
Closes #1014

### CODE REVIEW

**Comment by v1v** [comment]
I've just seen the below nodejs validations for >8.5:
- https://github.com/elastic/apm-agent-nodejs/blob/dd82e20e349e8ddd4f837bde46f12ad8b804c6d7/test/script/run_tests.sh#L59
- https://github.com/elastic/apm-agent-nodejs/blob/dd82e20e349e8ddd4f837bde46f12ad8b804c6d7/.ci/scripts/test_types_babel_esm.sh#L15

IIUC, the Node.js version 8 shouldn't run that particular npm command, should it?

File: test/instrumentation/modules/pg/knex.js [hunk 1]
```diff
@@ -14,11 +14,6 @@ var knexVersion = require('knex/package').version
 var pgVersion = require('pg/package').version
```
Comment (Qard) [inline test/instrumentation/modules/pg/knex.js]: This variable is not longer in-use and should be removed to pass the linting.

```suggestion
```

**Comment by Qard** [comment]
Other than the `pgVersion` linting issue, this LGTM, so feel free to land after fixing that.

File: test/instrumentation/modules/pg/knex.js (hunk 1 above)
Comment (watson) [reply test/instrumentation/modules/pg/knex.js]: Fixed

**Comment by watson** [comment]
@Qard I've added quite a few new things to the PR, so maybe you want to review once more?

**Pull request merged**